tools/fast/fastesting/basic.go

```diff
@@ -46,6 +46,10 @@ func NewTestEnvironment(ctx context.Context, t *testing.T, fastenvOpts fast.Envi
 	env, err := fast.NewEnvironmentMemoryGenesis(big.NewInt(1000000), dir, types.TestProofsMode)
 	require.NoError(err)
 
+	defer func() {
+		dumpEnvOutputOnFail(t, env.Processes())
```

Comment:
Should this be `TearDown` since it now calls this method?

Something I would like to see is, on test failure don't teardown completely instead leave the FAST repo in place with the stderr and stdout files and direct the user to that location. What do you think?

Reply:
No, because if we defer Teardown it will teardown the environment in this method which we don't want to do. We just want to dump this output if any of the `require.NoError` fail so we can see output of the setup steps in this function.

Reply:
What if we added a method like: `TeardownOnSuccess` that would conditionally teardown the environment?

Reply:
I think we should always teardown regardless if the test passes or fails. As mentioned else where I think we can preserve the logs to makes sure they are not removed in the event that a test fails though.

Reply:
Where is preserving logs on test failures being discussed? That's mainly what I am after here.

Reply:
I don't remember if there was an issue created yet, but @rosalinekarr wanted to have the daemon logs be stored in a `logs` directory alongside the project root such that they would 1) not be cleaned up on teardown, and also survive past restarts.

I think it was a discussion in slack a bit ago.